File: app/src/main/java/com/remotearthsolutions/expensetracker/databaseutils/models/AccountModel.java
package com.remotearthsolutions.expensetracker.databaseutils.models;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "account")
public class AccountModel {

    @NonNull
    @PrimaryKey(autoGenerate = true)
    private int id;

    @ColumnInfo(name = "name")
    private String name;

    @ColumnInfo(name = "icon")
    private String icon;

    @ColumnInfo(name = "amount")
    private double amount;

    @ColumnInfo(name = "notremovable")
    private int notremovable;

    @Ignore
    public AccountModel() {
    }

    public AccountModel(String name, String icon, double amount, int notremovable) {
        this.name = name;
        this.icon = icon;
        this.amount = amount;
        this.notremovable = notremovable;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }

    public int getNotremovable() {
        return notremovable;
    }

    public void setNotremovable(int notremovable) {
        this.notremovable = notremovable;
    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/databaseutils/models/AccountModel.kt
package com.remotearthsolutions.expensetracker.databaseutils.models

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity(tableName = "account")
class AccountModel {
    @PrimaryKey(autoGenerate = true)
    var id = 0
    @ColumnInfo(name = "name")
    var name: String? = null
    @ColumnInfo(name = "icon")
    var icon: String? = null
    @ColumnInfo(name = "amount")
    var amount = 0.0
    @ColumnInfo(name = "notremovable")
    var notremovable = 0

    @Ignore
    constructor()

    constructor(
        name: String?,
        icon: String?,
        amount: Double,
        notremovable: Int
    ) {
        this.name = name
        this.icon = icon
        this.amount = amount
        this.notremovable = notremovable
    }

}